src/core/mesh/qgsmeshplotter.h

```diff
@@ -0,0 +1,213 @@
+      SingleColor = 0,
+      //! Render with a color ramp
+      ColorRamp
+    };
+
+    //! Default constructor
+    QgsMeshStrokeColor() = default;
+    //! Constructor  with varying color depending on magnitude
+    QgsMeshStrokeColor( const QgsColorRampShader &colorRampShader );
+    //! Constructor  with fixed color
+    QgsMeshStrokeColor( const QColor &color );
+
+    //! Sets the color ramp to define the coloring
+    void setColor( const QgsColorRampShader &colorRampShader );
+
+    //! Sets the single color to define the coloring
+    void setColor( const QColor &color );
+
+    //! Returns the color corresponding to the magnitude
+    QColor color( double magnitude ) const;
+
+    //! Returns the coloring method used
+    QgsMeshStrokeColor::ColoringMethod coloringMethod() const;
+
+    //! Returns the color ramp shader
+    QgsColorRampShader colorRampShader() const;
+
+    //! Writes configuration to a new DOM element
+    QDomElement writeXml( QDomDocument &doc, const QgsReadWriteContext &context ) const;
+    //! Reads configuration from the given DOM element
+    void readXml( const QDomElement &elem, const QgsReadWriteContext &context );
+
+    /**
+     *  Returns the break values, graduated colors and the associated gradients between two values
+     *  - If the color is fixed or only one color for the interval (value1, value2), returns only one color in \a breakColors
+     *    and void lists for  \a breakValues, \a gradients
+     *  - If the color ramp is classified with 'exact', returns void \a gradients
+     *  - If the color ramp is classified with 'discrete', return \a gradients with uniform colors
+     *  - if nothing to render (out of range), return all lists void
+     */
+    void graduatedColors( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+
+  private:
+    QgsColorRampShader mColorRampShader;
+    QColor mSingleColor = Qt::black;
+
+    QgsMeshStrokeColor::ColoringMethod mColoringMethod = SingleColor;
+
+    QLinearGradient makeSimpleLinearGradient( const QColor &color1, const QColor &color2 ) const;
+
+    //! Returns the index of the color ramp shader width value immediatly inferior to value
+    int itemColorIndexInf( double value ) const;
+
+    void graduatedColorsExact( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+    void graduatedColorsInterpolated( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+    void graduatedColorsDiscrete( double value1, double value2, QList<double> &breakValues, QList<QColor> &breakColors, QList<QLinearGradient> &gradients ) const;
+};
+
+/**
+ * \ingroup core
+ *
+ * Class defining stroke width to render dataset. The width can vary depending on dataset value.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsMeshStrokeWidth
```

Comment:
QgsMeshStrokeWidth -> QgsMeshStrokeWidthSettings and move to the file with other mesh settings

for any calculation/logic I would suggest to move it to the mesh stroke/edge/1d renderer